lib/viewmodels/profile_view_model.dart

```diff
@@ -1,21 +1,52 @@
 import "package:hooks_riverpod/hooks_riverpod.dart";
 import "package:proxima/models/ui/profile_data.dart";
+import "package:proxima/services/database/user_repository_service.dart";
 import "package:proxima/viewmodels/login_view_model.dart";
 
 /// User profile view model
 class ProfileViewModel extends AsyncNotifier<ProfileData> {
   ProfileViewModel();
 
-  @override
-  Future<ProfileData> build() async {
+  Future<ProfileData> _fetchData() async {
     final user = ref.watch(userProvider).valueOrNull;
+    final userDataBase = ref.watch(userRepositoryProvider);
+    final uid = ref.watch(uidProvider);
+
     if (user == null) {
       return Future.error(
         "User must be logged in before displaying the home page.",
       );
     }
+    if (uid == null) {
+      return Future.error(
+        "User id was not found.",
+      );
+    }
+    final userData = await userDataBase.getUser(uid);
+
+    return ProfileData(loginUser: user, firestoreUser: userData);
+  }
+
+  @override
+  Future<ProfileData> build() async {
+    return _fetchData();
+  }
+
+  Future<void> addPoints(int points) async {
```

Comment:
Here I am not sure that this belongs to the view model as there will be no actual action in the profile page that will directly give you new centaury points. I think this functionality should probably stay only on the model and will be called by other actions. The centaury points will be changed based on more complex rules not yet defined.